### PR TITLE
Delete disable of library repos from spelling script

### PR DESCRIPTION
**Pull request by sarenameas**
- We updated the CI to check only the platform/ demos/ and integration-test/ directories.
- Add lexicon.txt to integration-test/ directory
- Adding each new library repo to the spellcheck script is not scalable and developers will forget to do so as new libraries are added. 

The test of this change in CI:
https://amazon-freertos-ci.corp.amazon.com/view/CSDK%20Jobs/job/csdk/job/spellcheck/1654/console

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: tools/spell/README.md [hunk 1]
```diff
@@ -13,9 +13,10 @@
 
 # How to create a lexicon.txt for a new library.
 
-1. Ensure there does not exist a file called "lexicon.txt" in your library's directory. Run the following command to create a lexicon.txt for your library:  
+1. Ensure there does not exist a file called "lexicon.txt" in your library's directory. Run the following command to create a lexicon.txt for your library:
    ```shell
-   find-unknown-comment-words -d <CSDK_ROOT>/libraries/<LIBRARY_TYPE>/<MY_LIBRARY_NAME> > <CSDK_ROOT>/libraries/<LIBRARY_TYPE>/<MY_LIBRARY_NAME>/lexicon.txt
```
Comment (sarenameas) [inline tools/spell/README.md:18]: These instructions were incorrect. This is because when the lexicon.txt using the redirection command, the script will intake in immediately during operation, therefore not properly creating a new one.

File: integration-test/lexicon.txt [hunk 2]
```diff
@@ -0,0 +1,66 @@
+ack
```
Comment (sarenameas) [inline integration-test/lexicon.txt:1]: There are words in here in preparation for the shadow integration tests.

File: tools/spell/find-unknown-comment-words [hunk 3]
```diff
@@ -109,15 +109,6 @@ do
...
         fi
-        # Disable spell check for submoduled libraries.
-        if [[ $file == *"coreMQTT"* ]]
-        then
-            continue
-        fi
-        if [[ $file == *"coreHTTP"* ]]
-        then
-            continue
-        fi
```
Comment (leegeth) [inline tools/spell/find-unknown-comment-words:120]: Why are these removed? The reason for these to be added was to disable spell check in submoduled repos. The spell check for those will be done in those repos.

Comment (sarenameas) [reply tools/spell/find-unknown-comment-words:120]: "We updated the CI to check only the platform/ demos/ and integration-test/ directories."

**Review by leegeth — APPROVED** [review]

**Review by dan4thewin — APPROVED** [review]
